app/models/external_activity.rb

```diff
@@ -151,6 +151,10 @@ def url(learner = nil)
       if learner
         append_query(uri, "learner=#{learner.id}") if append_learner_id_to_url
         append_query(uri, "c=#{learner.user.id}") if append_survey_monkey_uid
+        if append_auth_token
+          token = learner.user.create_access_token_valid_for(3.minutes)
+          append_query(uri, "token=#{token}")
+        end
```

Comment:
Not related to the code but do we have a cron job that deletes expired tokens?

Reply:
I don't believe we do. In other cases like this we've had the creation code check if there are any expired objects and delete them. That way we don't need a cron job, it does slow down the creation a bit, but this probably isn't a problem.

Reply:
I've now added `AccessGrant.prune!` before the token creation here, matching the pattern used in the AuthController.